### ISSUE TITLE
Synthetic streams accept zero schedule moduli: JAX n%0==0 silently destroys regime clocks (NaN targets, every-step changes)

### ISSUE THREAD

**Issue by ss251** [SlopDotCash/asi#143]
## Contribution provenance

- AI assistance: yes
- AI provider/model: xai / grok-4.6 (stream-surface sweep, reproductions, and fix in a local lane); anthropic / claude-fable-5 (independent verification and publish)
- Client / agent tooling: Grok Build CLI; claude-code
- Attribution status: self-reported

**Describe the bug**

Five synthetic streams accept a zero schedule modulus and silently destroy their own regime clock, because JAX int32 arithmetic does not raise on division by zero: `n % 0` is `0` and `n // 0` is negative.

- `PeriodicChangeStream(period=0)` emits `sin(2π·t/0) = NaN` **targets** against finite observations — every learner metric downstream is NaN with no error at the source.
- `CyclicStream(cycle_length=0)` indexes the wrong configuration.
- `SuttonExperiment1Stream` / `AbruptChangeStream(change_interval=0)` and `DynamicScaleShiftStream` change regime **every step** instead of on schedule.

A stream that mislabels its own regime corrupts every experiment run on it, and the failure is silent — the constructor accepts the value and the run completes. Sibling surfaces (`XDistShift`, `GauntletConfig`, feature-discovery `context_length`, closed-loop `phase_length`) already fail closed on the same class.

**To Reproduce** (at `origin/main` `da8b86cd`)

```
uv run python -c '
import jax.numpy as jnp, jax.random as jr
from alberta_framework.streams.synthetic import PeriodicChangeStream
p = PeriodicChangeStream(feature_dim=3, period=0, noise_std=0.0)
st = p.init(jr.key(0))
st, sample = p.step(st)
print("target:", sample.target)     # nan — accepted, no error
print("jax semantics:", jnp.int32(5) % jnp.int32(0), jnp.int32(5) // jnp.int32(0))
'
```

**Expected behavior**

Zero (and negative/non-int) schedule moduli are rejected at construction with a named error, matching the fail-closed siblings. Positive schedules are bit-identical.

**Additional context**

Fix PR follows immediately: a shared `_require_positive_int` applied at all five constructors, parameterized rejection regressions across every class plus accepts-valid controls, and a deterministic finite-trajectory probe confirming valid schedules are bit-identical.
